MongoDbGenericRepository/DefaultBaseMongodbRepository.TKey.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections.Generic;
+using MongoDB.Driver;
+using Mongodb.Driver.Extensions.Abstractions;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using MongoDbGenericRepository;
+using MongoDbGenericRepository.Models;
+
+namespace Mongodb.Driver.Extensions
+{
+    public abstract class DefaultBaseTKeyMongodbRepository<TDocument, TKey> : BaseMongoRepository<TKey>, IDefaultBaseTKeyMongodbRepository<TDocument, TKey>
```

Comment:
Purpose?

Reply:
Easier to Use BaseMongodbRepository
and InitRepository with no params in ctor
```
        protected abstract string MongodbConnectionString { get; }

        protected virtual string DefaultCollectionName => GetDefaultCollectionName<TDocument>();
```

Reply:
but.. an throwing an argument exception, if it's null. Meaning it cannot work on it's own like that.

Reply:
I looks like you're forcing your custom implementation for your needs into something that's generic, and repo inheriting from BaseMongoRepository is custom and should be kept in you code. 
It's not generic enough. 
I cannot simply accept everyone's implementation inheriting from BaseMongoRepository.
It's great that you're building something custom based on BaseMongoRepository for your needs, but most of that functionality is for your application only.